### ISSUE TITLE
4k support at MBA early 2014 HD5000

### ISSUE THREAD

**Issue by k-Tracker** [Floris497/mac-pixel-clock-patch#3]
Hi Floris497, 
I saw the number of reports of successful 4k enabling at MBA with HD4xxx graphics, but read that HD5000 has a driver issue on Yosemite preventing that. Can you please advise if that's the case and you successfully tried HD5000 for 4k at 30Hz? 
I have applied the script at 10.10.4 and have Switchresx installed but can't install 4k mode there. Any advice is appreciated.

**Comment by Floris497** [comment]
I have the same MBA from 2013 (exactly the same) getting it to 4k is "possible" but involves editing of the framebuffer kernel extension an makes your laptop "unusable" witout an external display and unstable with external display. I would not recommend doing it. Since im out of the country i have no acces to my research on the subject. I will post some info here when i'm back. In about a week.

**Comment by k-Tracker** [comment]
Thank you, I'll be very interested to learn more. But of course I wish to keep a local screen too, or at least be able to switch between them ))

**Comment by Floris497** [comment]
Look at my post from dec 22 https://code.google.com/p/mac-pixel-clock-patch/wiki/Documentation

**Comment by k-Tracker** [comment]
Thank you, 
looks, I did exactly how you recommended for patching frame buffer, and here is the confirmation after the restart bellow. Did you enable 4k in a switchresX after these changes? 

 AAPL,ig-platform-id: 0x0a260006 (Haswell ULT Mobile GT3) found @ 0x481a0/295328
Source file: /System/Library/Extensions/AppleIntelFramebufferAzul.kext/Contents/MacOS/AppleIntelFramebufferAzul
--------------------------------------------------------------------------
000481a0: 0600 260a 0103 0303 0000 0002 0000 8001 (32 MB BIOS-allocated memory, 24 MB frame buffer memory)
000481b0: 0000 6000 0000 0060 d90a 0000 d90a 0000 (6 MB cursor bytes, 1536 MB VRAM, BCL freq. 2777 Hz, max. BCL PWM 2777 Hz)
000481c0: 0000 0000 0000 0000 0000 0800 0200 0000 (port 0, LVDS connector)
000481d0: 3000 0000 0105 0900 0004 0000 8700 0000 (port 5, DisplayPort connector)
000481e0: 0204 0900 0004 0000 8700 0000 ff00 0000 (port 6, DisplayPort connector / port unused ...
000481f0: 0100 0000 4000 0000 0f00 0000 0101 0000  VGA connector)
00048200: 0400 0000 0000 0000 0e00 0000 0000 0000

**Comment by Floris497** [comment]
Yep, i have no idea if it still works, as noted there is a small black bar at the bottom, using a slightly lower resolution deals with that, if your display can handle all frequencies between 30 an 60 you should see if you can get it to 40Hz or even a little heigher. Does it work for you?

**Comment by k-Tracker** [comment]
I'll try... I supposed, I physically can't transfer the 4k at 30Hz+ with a DP1.1. May I ask you to make a screenshot of the SwitchResX, I'll match the parameters at least...

**Comment by Floris497** [comment]
I only have my phone with me.. So I can't give you a screenshot, but set it on 3840x2160 on 30Hz should be fine, try to calculate some smaller resolutions 3440x(??) Or something make sure it is the same ratio. If your screen limits it you can't go heigher then 30Hz but if your screen allows it your computer should be able to do it.

**Comment by AndreyKudryavtsev** [comment]
Hi Floris497, 
I have upgraded to El Capitan and reinstalled the patch recently. However I still can't enable 4k on MBA early2014. It's connected to Asus PB287Q over HDMI by Accell B086B-008b-2. 
If you have similar configuration enabled, can you please post the screenshot of Switchres settings? I appreciate any other tip to get it running.

**Comment by Floris497** [comment]
I Don't think 4k is possible on the MacBook Air 2014, it's about the same hardware wise as the 2013 model i have IIRC but it is possible to get resolution close to 4k, look in the wiki of the original repository for my success and failure stories.. There are some explanations from me how to edit your frambuffer kext.. But I certainly don't recommend to fo so.. I have no experience with active adapters, and thus have no idea if they make 4k possible.. Hope this helped you, I think I might try to give it a go on my MBA with 10.11.. I'll let you hear if i try.. Hope this information helps you!

**Comment by AndreyKudryavtsev** [comment]
Thank you Floris, 
did you mean following original Wiki? 
https://code.google.com/p/mac-pixel-clock-patch/wiki/Documentation
My question was not answered there and now the project is archived... Appreciate if you can provide some guidance. I'm willing to try )))
I broke the kext few times already, booting in recovery is not a problem.

**Comment by Floris497** [comment]
Okay great, I'll let you know tomorrow!
Look at the comments (22 dec 2014, that is me)
